itest/assertions.go

```diff
@@ -92,55 +94,90 @@ func assetScriptKeyIsLocalCheck(isLocal bool) assetCheck {
 	}
 }
 
+// sortAssetsByName converts an unordered list of assets to a map of lists of
+// assets, where all assets in a list have the same name.
+func sortAssetsByName(assets []*taprpc.Asset) map[string][]*taprpc.Asset {
+	assetLists := make(map[string][]*taprpc.Asset)
+	for _, a := range assets {
+		if _, ok := assetLists[a.AssetGenesis.Name]; !ok {
+			assetLists[a.AssetGenesis.Name] = []*taprpc.Asset{}
```

Comment:
I don't think we need to pre-initialize this, since `append()` will work even if the first argument is `nil`.